src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -827,56 +826,34 @@ private static StringValue SubstituteWorker(EvalVisitor eval, IRContext irContex
                 return source;
             }
 
-            var sourceValue = source.Value;
-            var idx = sourceValue.IndexOf(match.Value);
+            StringBuilder strBuilder = new StringBuilder(source.Value);
             if (instanceNum < 0)
             {
-                while (idx >= 0)
-                {
-                    eval.CheckCancel();
-
-                    var temp = sourceValue.Substring(0, idx) + replacement.Value;
-                    sourceValue = sourceValue.Substring(idx + match.Value.Length);
-                    var idx2 = sourceValue.IndexOf(match.Value);
-                    if (idx2 < 0)
-                    {
-                        idx = idx2;
-                    }
-                    else
-                    {
-                        idx = temp.Length + idx2;
-                    }
-
-                    sourceValue = temp + sourceValue;
-                }
+                strBuilder.Replace(match.Value, replacement.Value);
```

Comment:
>Replace [](http://example.com/codeflow?start=27&length=7)

This is a big win - .Net's Replace() will handle for us.

Reply:
yes it handles all for us.